users/views_test.py

```diff
@@ -15,21 +16,34 @@ def test_get_user_by_id(user_client, user):
         "username": user.username,
         "email": user.email,
         "name": user.name,
+        "is_anonymous": False,
+        "is_authenticated": True,
         "created_on": drf_datetime(user.created_on),
         "updated_on": drf_datetime(user.updated_on),
     }
 
 
-def test_get_user_by_me(user_client, user):
+@pytest.mark.parametrize("is_anonymous", [True, False])
+def test_get_user_by_me(client, user, is_anonymous):
     """Test that user can request their own user by the 'me' alias"""
-    resp = user_client.get(reverse("users_api-detail", kwargs={"pk": "me"}))
+    if not is_anonymous:
```

Comment:
Take or leave: you could parametrize the `client` param with a logged-in and not-logged-in client, and in the body of the test, check whether the client is logged in via utility method (`client.cookies.get("sessionid") is not None`). You'd have to rename `client` to something like `test_client`

Reply:
I think I'll leave this be since I feel that binds this test to the internals of the django test client.